etl/steps/data/garden/animal_welfare/2023-08-14/number_of_farmed_fish.py

```diff
@@ -0,0 +1,117 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from owid.catalog import Dataset, Table
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Column names to select and how to rename them.
+COLUMNS = {
+    "country": "country",
+    "year": "year",
+    "estimated_numbers__millions__lower": "n_farmed_fish_low",
+    "estimated_numbers__millions__upper": "n_farmed_fish_high",
+}
+
+# Regions to create aggregates for.
+REGIONS_TO_ADD = [
+    "North America",
+    "South America",
+    "Europe",
+    "European Union (27)",
+    "Africa",
+    "Asia",
+    "Oceania",
+    "Low-income countries",
+    "Upper-middle-income countries",
+    "Lower-middle-income countries",
+    "High-income countries",
+]
+
+
+def add_region_aggregates(tb: Table, ds_regions: Dataset, ds_income_groups: Dataset) -> Table:
+    tb = tb.copy()
+    for region in REGIONS_TO_ADD:
+        # List of countries in region.
+        countries_in_region = geo.list_members_of_region(
+            region=region,
+            ds_regions=ds_regions,
+            ds_income_groups=ds_income_groups,
+        )
+
+        # Add region aggregates.
+        tb = geo.add_region_aggregates(
+            df=tb,
+            region=region,
+            countries_in_region=countries_in_region,
+            countries_that_must_have_data=[],
+            # Here we allow aggregating even when there are few countries informed.
+            # However, if absolutely all countries have nan, we want the aggregate to be nan, not zero.
+            frac_allowed_nans_per_year=0.999,
+            num_allowed_nans_per_year=None,
+        )
+    return tb
+
+
+def run_sanity_checks_on_outputs(tb: Table) -> None:
```

Comment:
Maybe a check like this should be added to geo, or inside the `add_region_aggregates` function. We would need to specify whihc regions form the world, though.

Reply:
Yes, that could be an optional check. It may not always be fulfilled (because of rounding errors, or if you use an aggregate that is not `sum`). But it most cases it would be useful.